akka-stream/src/main/scala/akka/stream/io/SynchronousFileSource.scala

```diff
@@ -25,8 +25,7 @@ object SynchronousFileSource {
    * It materializes a [[Future]] containing the number of bytes read from the source file upon completion.
    */
   def apply(f: File, chunkSize: Int = DefaultChunkSize): Source[ByteString, Future[Long]] =
-    new Source(new SynchronousFileSource(f, chunkSize, DefaultAttributes, Source.shape("SynchronousFileSource")))
-      .named(DefaultAttributes.nameOption.get)
+    new Source(new SynchronousFileSource(f, chunkSize, DefaultAttributes, Source.shape("SynchronousFileSource")).wrap()) // TO DISCUSS: I had to add wrap() here to make the name available
```

Comment:
@drewhk But why do I need to call wrap here if I removed the `.named(DefaultAttributes.nameOption.get)` section?

I fear that remembering where and when `wrap()` needs to be called will lead to very subtle bugs.